tests/readfile-context.nix

```diff
@@ -10,8 +10,17 @@ let
     input = "${input}/hello";
   };
 
+  discardableDependent = mkDerivation {
+    name = "dependent-noref";
+    builder = ./readfile-context.builder.sh;
+    input = "read-dependent"; # No reference in it
+  };
+
   readDependent = mkDerivation {
-    name = "read-dependent";
+    # Will evaluate correctly because `readFile
+    # discardableDependent` won't keep any context (since it will
+    # refscan its content, and this one doesn't have any reference)
+    name = builtins.readFile discardableDependent;
```

Comment:
Apparently this succeeds even if there's a context.

Reply:
How did you see that? I tried replacing `discardableDependent` by `dependent` and the evaluation fails because `the string '/tmp/nix-test/tests/readfile-context/store/dk4iy6vpksymyjpm1wl0646hh5p3vm3c-simple/hello' is not allowed to refer to a store path `.

I also tried with `builtins.substring 0 0 (builtins.readFile dependent) + "foo"` just to make sure that I had the same error with a valid `name`, and I indeed have it

Reply:
Ah, I see, the problem is that `discardableDependent` has no references at all so the test succeeds even without this patch.

What we should do to test this properly is have a derivation that writes *two* files, one with a reference and one without, and read the one without a reference.

Reply:
Oh yes, of course! I assumed that `discardableDepend` would refer to itself, but it isn't by default

Reply:
Updated the test.